misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/utils/VariableUtils.java

```diff
@@ -132,6 +114,60 @@ public static Value getVariableValue(SuspendedContext context, String name) thro
         }
     }
 
+    /**
+     * Returns runtime value of the matching global variable, for the given name.
+     *
+     * @param context       suspended context
+     * @param nameReference name of the variable to be retrieved
+     * @return the JDI value instance of the global variable
+     */
+    private static Optional<BExpressionValue> searchGlobalVariables(SuspendedContext context, String nameReference) {
+        String classQName = PackageUtils.getQualifiedClassName(context, INIT_CLASS_NAME);
+        return getFieldValue(context, classQName, nameReference);
+    }
+
+    /**
+     * Returns runtime value of the matching global variable, for the given name.
+     *
+     * @param moduleSymbol  module symbol retrieved from the semantic API
+     * @param nameReference name of the variable to be retrieved
+     * @return the JDI value instance of the global variable
+     */
+    public static Optional<BExpressionValue> searchModuleVariables(SuspendedContext context, ModuleSymbol moduleSymbol,
+                                                                   String nameReference) {
+        String classQName = getInitClassName(moduleSymbol);
+        return getFieldValue(context, classQName, nameReference);
+    }
+
+    /**
+     * Returns full-qualified name of the init class, for a given Ballerina module.
+     *
+     * @param moduleSymbol module symbol retrieved from the semantic API
+     * @return full-qualified class name
+     */
+    private static String getInitClassName(ModuleSymbol moduleSymbol) {
+        StringJoiner classNameJoiner = new StringJoiner(".");
+        classNameJoiner.add(moduleSymbol.id().orgName())
+                .add(encodeModuleName(moduleSymbol.id().moduleName()))
+                .add(moduleSymbol.id().version().split("\\.")[0])
```

Comment:
Refer this [comment](https://github.com/ballerina-platform/ballerina-lang/pull/32621/files#r704925279)